### PR TITLE
feat: [PL-41778]: Add KMS integration for bijou

### PR DESCRIPTION
**Pull request by AnupamIO**
Testing:
1- Tested for GCP kms
2- Tested for AWS kms with access key and secet key

Note: mapstruct and mapstruct_proto dependencies has to be in specific order (proto first) other we will get issue.
https://github.com/entur/mapstruct-spi-protobuf/issues/232

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56235)

### CODE REVIEW

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -287,7 +287,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
+shouldConfigureWithPMS: true
```
Comment (xingchi-jin) [inline 120-ng-manager/config.yml:290]: is this intentional change ?

File: 260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel [hunk 2]
```diff
@@ -16,6 +16,8 @@ java_library(
         "//955-delegate-beans/src/main/proto/io/harness/delegate/core/beans:955_delegate_core_task_java_proto",
         "//970-api-services-beans/src/main/java/io/harness/security/encryption:module",
         "//999-annotations/src/main/java/io/harness/annotations/dev:module",
+                "//:mapstruct_proto",
```
Comment (xingchi-jin) [inline 260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel:19]: indentation

File: 120-ng-manager/config.yml (hunk 1 above)
Comment (AnupamIO) [reply 120-ng-manager/config.yml:290]: just for testing, will remove this.

File: 260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel (hunk 2 above)
Comment (AnupamIO) [reply 260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel:19]: ack

**Comment by VikasMaddukuriHarness** [comment]
Please remove the changes from the NgDelegate2TaskExecutor.
Please let me know once you merge this. I will rebase and merge my pr related to vault mappers
@AnupamIO @xingchi-jin

File: 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/secret_details.proto [hunk 3]
```diff
@@ -137,7 +137,7 @@ message AwsSecretsManagerConfig {
 message GcpKmsConfig {
   string project_id = 1;
   string region = 2;
-  string king_ring = 3;
```
Comment (xingchi-jin) [inline 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/secret_details.proto:140]: nice catch : p

**Review by xingchi-jin — APPROVED** [review]

**Review by VikasMaddukuriHarness — DISMISSED** [review]

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger uts

**Review by mdmilic — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56235)* status: 14 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel` line 15 at r5](https://reviewable.io/reviews/harness/harness-core/56235#-NkzECtK5-qtf0feM6uv:-NkzECtK5-qtf0feM6uw:bx6uan8) ([raw file](https://github.com/harness/harness-core/blob/1ee99e0334182a00b38754cdeb1da6ddb5da87b9/260-delegate/src/main/java/io/harness/delegate/service/secret/BUILD.bazel#L15)):*
> ```Starlark
>     deps = [
>         "//:mapstruct_proto",
>         "//:mapstruct",
> ```

Not sure if you need both deps or just mapstruct_proto (which would include the original mapstruct I think), but also doesn't make any difference either as they would both be used.

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger ut5

**Comment by AnupamIO** [comment]
trigger ut9

**Review by mdmilic — APPROVED** [review]
> Reviewed 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56235)* status: 14 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Pull request merged**